ludwig/api.py

```diff
@@ -242,6 +241,7 @@ def train(
         skip_save_progress: bool = False,
         skip_save_log: bool = False,
         skip_save_processed_input: bool = False,
+        skip_save_inference_module: bool = False,
```

Comment:
Good question– in `ludwig/models/trainer.py:1076` of this PR, we make a best-effort attempt at saving the model as an inference module. If it fails, then the error is logged and saving the inference module is skipped. Regardless of the outcome, the usual LudwigModel is saved.

In `ludwig/contribs/mlflow/model.py`, we check if the inference module was successfully saved. If not, then we use the usual LudwigModel.

Because of this logic, the app should save/load inference modules whenever we support the features used during training. Otherwise, it should fail gracefully into our previous set up.

Does that seem reasonable? WDYT?

Reply:
If we elegantly handle when inference module saving/loading fails, then by default setting skipping to `False` seems fine to me. Thanks for the detailed explanation of the robustness of the behavior!